### PR TITLE
Separated HTML Files from Extension.ts

### PR DESCRIPTION
**Pull request by marzzuki**
- Code is very inefficient as it converts the file buffer to string then it replaces `{sfs}` to the `stackFrameArray` every time the function is called

- HTML can't link to external CSS and JS files yet, so they have to be embedded using `<style>` and `<script>` tags

### CODE REVIEW

File: media/visual.html [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    <div id="stack"></div>
+</div>
+
+
+<h2 class = "center">Stack Visualization (Tree View)</h2>
+<div id="trees-container">
+</div>
+
+<sf id="sf-id" data-frames='{sfs}' ></sf>
+<script src="D:/Projects/Programming/callstackninja/media/visualize.js"></script>
```
Comment (marzoukali) [inline media/visual.html]: this path should be added as reference path not absolute. ../../xx.js

File: src/extension.ts [hunk 2]
```diff
@@ -48,191 +58,18 @@ export function activate(context: vscode.ExtensionContext) {
...
-				treesContainer.innerHTML = fullTreeStructure;
-		
-			}
-			*/
-		</script>
-		</body>
-		</html>`;
-
-	
+		return fs.readFileSync(extensionPath + "/media/index.html").toString();
```
Comment (marzoukali) [inline src/extension.ts]: use String interpolation instead of +

File: media/visual.html [hunk 3]
```diff
@@ -0,0 +1,163 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="D:/Projects/Programming/callstackninja/media/visual.css">
```
Comment (marzoukali) [inline media/visual.html]: use reference path instead of absolute

Comment (marzoukali) [inline media/visual.html]: Is this used any where?

File: src/extension.ts [hunk 4]
```diff
@@ -48,191 +58,18 @@ export function activate(context: vscode.ExtensionContext) {
...
+	}
+	else{
+		//very inefficient
```
Comment (marzoukali) [inline src/extension.ts:75]: Could we refer to this https://code.visualstudio.com/api/extension-guides/webview#passing-messages-from-an-extension-to-a-webview to see if we can pass the variable to the webview this way ?

**Review by marzoukali — CHANGES_REQUESTED** [review]

File: media/visual.html (hunk 1 above)
Comment (marzzuki) [reply media/visual.html]: It is solved using `vscode.Uri` instead of a reference path.

File: src/extension.ts (hunk 2 above)
Comment (marzzuki) [reply src/extension.ts]: Done

File: media/visual.html (hunk 3 above)
Comment (marzzuki) [reply media/visual.html]: It is solved using vscode.Uri instead of a reference path.

Comment (marzzuki) [reply media/visual.html]: What do you mean?

File: src/extension.ts (hunk 4 above)
Comment (marzzuki) [reply src/extension.ts:75]: Working on it

**Pull request merged**